### DISCUSSION TITLE
Confirm Steam market listings

### DISCUSSION THREAD

**Discussion by dapal-003** [ValvePython/steamctl#34] [Q&A] [answered]
So I recently switched over to using steamctl to generate my authentication codes, but alongside that came the fact that I couldn't re-authenticate with my phone. No biggie I thought to myself but now I've come to the problem of wanting to list an item on the steam market, but being unable to confirm those listings on mobile as the authentication is not associated. 

Is there a way to confirm these actions through steamctl?

**Comment by rossengeorgiev** [comment, marked as answer]
`steamctl` doesn't have a command to do that yet, but when adding an authenticator you can set it up so that both `Steam Mobile App` and `steamctl` work. Just follow the prompts.

**Reply by dapal-003** [reply]
Thanks for confirming that, when I set up `steamctl` it wouldn't allow me to set up the authentication for both `Steam Mobile App` and `steamctl`. The sms code I received would not get accepted inside of the app as it would claimed it was invalid every single time.

Glad to see that it's something to be implemented later on though!

**Reply by rossengeorgiev** [reply]
That doesn't sound right. The process goes like so:

1. `steamctl authenticator add <username>`
2. Do you want to use Steam app? = yes
3. Ignore first SMS code
3. Open `Steam Mobile app`, login with the same username, and select `Add Authenticator`
4. Enter second SMS code that you received
5. Save recovery code, and wait 30s
6. Generate a code with `Steam Mobile app` and enter on `steamctl` prompt
7. If everything went ok, you should get `Authenticator added successfully!`
8. Now you can use both apps

**Reply by dapal-003** [reply]
Ah, that was what went wrong. I used the first sms code inside of the second when trying to add the authenticator on the `Steam Mobile app` Following this step by step allowed me to use both, thanks again for developing this.

**Comment by dapal-003** [comment]
A workaround I found in order to still accept these confirmations is to use [ASF](https://github.com/JustArchiNET/ArchiSteamFarm), using the .json generated by `steamctl` it is possible to add authentication to an `ASF` bot, and from there accept the trade offers within `ASF` with the command `2faok [bots]`
